torch/_inductor/codecache.py

```diff
@@ -893,6 +893,8 @@ def get_include_and_linking_paths(
             # and raise error together with instructions at compilation error later
         else:
             libs = ["omp"] if config.is_fbcode() else ["gomp"]
+            if config.assert_indirect_indexing:
+                libs += ["c10", "torch_cpu"]
```

Comment:
@jgong5 @peterbell10 @jansel The PR currently has some issues that we are not currently linking against `torchlib`, as the kernels cannot find `TORCH_CHECK`. Is it alright to unconditionally import these? This is in an `else:` block at the moment, but should probably be done for most configs if `assert_indirect_indexing`.

Reply:
Not sure if we might run into ABI issues here, I'll just add that we only need `c10` for `TORCH_CHECK`.

Reply:
note that we already add `libs += ["c10_cuda", "cuda", "torch_cuda"]` for CUDA.

Reply:
> Is it alright to unconditionally import these?

Yes, sounds good to me.